flow/project.py

```diff
@@ -4089,11 +4032,24 @@ def hello(job):
         if isinstance(func, str):
             return lambda op: cls.operation(op, name=func)
 
+        if func in chain(
+            *cls._OPERATION_PRECONDITIONS.values(),
+            *cls._OPERATION_POSTCONDITIONS.values(),
+        ):
+            raise ValueError("A condition function cannot be used as an operation")
+
```

Comment:
These changes seem somewhat unrelated and are not covered by the changelog entry. I assume this is in here due to historical reasons? I might suggest to split this off cleanly.

Reply:
We can do that, I didn't mention this in the changelog because we already make these API promises, and there is not an issue relating to this.

Reply:
Sure, let's just split these changes into a separate PR for cleanliness.